### ISSUE TITLE
Memory leak with BigDecimal::EXCEPTION_OVERFLOW

### ISSUE THREAD

**Issue by MaxLap** [ruby/bigdecimal#290]
Hello, the following causes a memory leak.

I left a commented line, both trigger an memory leak. I assume it's the same bug

```
require 'bigdecimal'

BigDecimal.mode(BigDecimal::EXCEPTION_OVERFLOW, true)

10.times do |j|
  100000.times do |i|
    begin
      BigDecimal("1E11111111111111111111")
      #BigDecimal("1E1000000000000000000")**10 # Also leaks
    rescue
    end
  end

  GC.start
  puts `ps -o rss= -p #{$$}`.to_i
end
```

```
$ ruby local.rb
28652
34988
41060
47396
53732
59804
66140
72476
78548
84884
```

Happens in 3.2.3, 3.3.0, 3.3-head, ruby-head (of a few days ago, which use big-decimal 3.1.6).

**Comment by mrkn** [comment]
@MaxLap Thank you for your report. I'm fixing this bug in #294.

**Comment by mrkn** [comment]
@MaxLap I've released v3.1.8.  It should fix this problem.
